astxmlparser.cpp

```diff
@@ -216,49 +214,81 @@ Data::SourceLocation AstXmlParser::readLocationFromAttributes()
 
 namespace
 {
-Data::Type mapNameToDataType(const QStringRef &name)
-{
-    static const QMap<QString, Data::Type> mapping = {
-        { "BooleanType",       Data::Type::Boolean },
-        { "NullType",          Data::Type::Null },
-        { "IntegerType",       Data::Type::Integer },
-        { "RealType",          Data::Type::Real },
-        { "BitStringType",     Data::Type::BitString },
-        { "OctetStringType",   Data::Type::OctetString },
-        { "IA5StringType",     Data::Type::IA5String },
-        { "NumericStringType", Data::Type::NumericString },
-        { "EnumeratedType",    Data::Type::Enumerated },
-        { "ChoiceType",        Data::Type::Choice },
-        { "SequenceType",      Data::Type::Sequence },
-        { "SequenceOfType",    Data::Type::SequenceOf },
+Data::Type::Kind mapNameToDataType(const QStringRef &name)
+{
+    static const QMap<QString, Data::Type::Kind> mapping = {
+        { "BooleanType",       Data::Type::Kind::Boolean },
+        { "NullType",          Data::Type::Kind::Null },
+        { "IntegerType",       Data::Type::Kind::Integer },
+        { "RealType",          Data::Type::Kind::Real },
+        { "BitStringType",     Data::Type::Kind::BitString },
+        { "OctetStringType",   Data::Type::Kind::OctetString },
+        { "IA5StringType",     Data::Type::Kind::IA5String },
+        { "NumericStringType", Data::Type::Kind::NumericString },
+        { "EnumeratedType",    Data::Type::Kind::Enumerated },
+        { "ChoiceType",        Data::Type::Kind::Choice },
+        { "SequenceType",      Data::Type::Kind::Sequence },
+        { "SequenceOfType",    Data::Type::Kind::SequenceOf },
```

Comment:
damn it's ugly :D "Type::Kind" - both words are quite similar and it makes it a little unreadable. Maybe Type should be a base class? Inheritance is better than enums..